src/HTTPRouter.jl

```diff
@@ -288,15 +291,15 @@ function with_cors!(response::HTTP.Response)
     response
 end
 
-function with_cacheable!(response::HTTP.Response)
-    second = 1
-    minute = 60second
-    hour = 60minute
-    day = 24hour
-    year = 365day
-
-    HTTP.setheader(response, "Cache-Control" => "public, max-age=$(10year), immutable")
-    response
+function with_cacheable!(max_age::Int)
+    return (response::HTTP.Response) -> begin
+        if max_age > 0
+            HTTP.setheader(response, "Cache-Control" => "public, max-age=$(max_age), immutable")
```

Comment:
Not sure if you want `immutable` if you set a small custom value... Maybe only add `immutable` if the value is at the default setting (10yrs)?